2010-01-17  Samuel Thibault  <samuel.thibault <at> ens-lyon.org>

	* sysdeps/mach/hurd/bits/libc-lock.h (__rtld_lock_recursive_t): New
	type.
	(__rtld_lock_initialize): New macro.

---
 libc-lock.h |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/sysdeps/mach/hurd/bits/libc-lock.h b/sysdeps/mach/hurd/bits/libc-lock.h
index 0fa90bc..eadd98d 100644
--- a/sysdeps/mach/hurd/bits/libc-lock.h
+++ b/sysdeps/mach/hurd/bits/libc-lock.h
@@ -31,6 +31,7 @@ typedef struct
   void *owner;
   int count;
 } __libc_lock_recursive_t;
+typedef __libc_lock_recursive_t __rtld_lock_recursive_t;

 #define __libc_lock_owner_self() ((void *) __hurd_threadvar_location (0))

@@ -121,6 +122,8 @@ typedef struct __libc_lock_recursive_opaque__ __libc_lock_recursive_t;

 #define __rtld_lock_init_recursive(NAME) \
   __libc_lock_init_recursive (NAME)
+#define __rtld_lock_initialize(NAME) \
+  (void) ((NAME) = (__rtld_lock_recursive_t) _RTLD_LOCK_RECURSIVE_INITIALIZER)
 #define __rtld_lock_trylock_recursive(NAME) \
   __libc_lock_trylock_recursive (NAME)
 #define __rtld_lock_lock_recursive(NAME) \
